File: MyFrame/mvcframe/src/main/java/xmx/zs/mvcframe/receiver/NetWorkReceiver.java
package xmx.zs.mvcframe.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;

import xmx.zs.mvcframe.utils.NetWorkUtils;





/*
 * @创建者     mqm
 * @博客       http://blog.csdn.net/u012792686
 * @创建时间   2017/5/16
 * @本类描述	  监听手机网络状态广播
 * @内容说明  当手机状态发生改变,将数据用接口回调方式,传递出去
 * @补充内容
 *
 * ---------------------------------     
 * @更新时间   
 * @新增内容   
 *
 */

public class NetWorkReceiver extends BroadcastReceiver {

    private String mNetWorkTypeName;


    @Override
    public void onReceive(final Context context, Intent intent) {
        String action = intent.getAction();
        if (action.equals(ConnectivityManager.CONNECTIVITY_ACTION)) {
            //获取网络状态
            mNetWorkTypeName = NetWorkUtils.getNetWorkTypeName(context);
            if (mOnNetWorkStataChangedListener != null) {

                mOnNetWorkStataChangedListener.onNetWorkStataChanged(mNetWorkTypeName);
            }

        }
    }

    public interface onNetWorkStataChangedListener {
        void onNetWorkStataChanged(String mNetWorkTypeName);
    }

    private onNetWorkStataChangedListener mOnNetWorkStataChangedListener;

    public void setOnNetWorkStataChangedListener(onNetWorkStataChangedListener listener) {
        this.mOnNetWorkStataChangedListener = listener;
    }
}

File: MyFrame/mvpframe/src/main/java/zs/xmx/mvpframe/receiver/NetWorkReceiver.kt
package zs.xmx.mvpframe.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager

import zs.xmx.mvpframe.utils.NetWorkUtils


/*
 * @创建者     mqm
 * @博客       http://blog.csdn.net/u012792686
 * @创建时间   2017/5/16
 * @本类描述	  监听手机网络状态广播
 * @内容说明  当手机状态发生改变,将数据用接口回调方式,传递出去
 * @补充内容
 *
 * ---------------------------------
 * @更新时间
 * @新增内容
 *
 */

class NetWorkReceiver : BroadcastReceiver() {

    private var mOnNetWorkStateChangedListener: onNetWorkStateChangedListener? = null


    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action
        if (action == ConnectivityManager.CONNECTIVITY_ACTION) {
            //获取网络状态
            val netWorkTypeName = NetWorkUtils.getNetWorkTypeName(context)
            if (mOnNetWorkStateChangedListener != null) {

                mOnNetWorkStateChangedListener!!.onNetWorkStateChanged(netWorkTypeName)
            }

        }
    }

    interface onNetWorkStateChangedListener {
        fun onNetWorkStateChanged(mNetWorkTypeName: String)
    }

    fun setOnNetWorkStateChangedListener(listener: onNetWorkStateChangedListener) {
        this.mOnNetWorkStateChangedListener = listener
    }
}
